File: 2018B/practiceA/solution.go
// package main

// import (
// 	"fmt"
// 	"math"
// )

// func noNine(F, L int) int {
// 	var res int
// 	for i := F; i <= L; i++ {
// 		if i%9 == 0 {
// 			continue
// 		}
// 		s := i
// 		t := 0
// 		for s > 0 {
// 			// fmt.Println(t, s, "t, s")
// 			t = s % 10
// 			s = s / 10
// 			// fmt.Println(t, s, "t, s")
// 			if t == 9 {
// 				res--
// 				break
// 			}
// 		}
// 		res++
// 		// fmt.Println(res, "res")
// 	}
// 	return res
// }

package main

import (
	"fmt"
)

func noNine(t int) int {
	var A []int
	var digit [18]int
	var res int
	var n = t
	for i := 1; i < 18; i++ {
		digit[0] = 1
		digit[i] = digit[i-1] * 9
	}
	for n > 0 {
		i := n % 10
		n = n / 10
		A = append(A, i)
	}
	for i := len(A) - 1; i > 0; i-- {
		res += A[i] * digit[i-1] * 8
		if A[i] == 9 {
			return res
		}
	}
	t -= A[0]
	for i := 0; i <= A[0] && i != 9; i++ {
		if ((t + i) % 9) > 0 {
			res++
		}
	}
	return res
}

func main() {
	var T, F, L int
	fmt.Scan(&T)
	for i := 0; i < T; i++ {
		fmt.Scan(&F)
		fmt.Scan(&L)
		s := noNine(L) - noNine(F) + 1
		fmt.Printf("Case #%d: %d\n", i+1, s)
	}
}

File: 2018B/practiceA/solution.cpp
#include <bits/stdc++.h>
using namespace std;
#define LL long long
LL pn[17] = {1};

LL noNine(LL x) {
  vector<int> num;
  LL t = x, res = 0;
  if(t == 0) {
    num.push_back(0);
  }
  while (t)
  {
    num.push_back(t%10);
    t /= 10;
  }
  int len = num.size();
  for(int i = len - 1; i > 0; i--) {
    res += num[i] * pn[i - 1] * 8;
    if(num[i] == 9) {
      return res;
    }
  }
  x -= num[0];
  for(int i = 0; i <= num[0] && i != 9; i++) {
    if((x + i) % 9) {
      res++;
    }
  }
  return res;
}

int main() {
  int T, o = 0;
  for(int i = 1; i < 18; i++) {
    pn[i] = pn[i - 1]*9;
  }
  cin >> T;
  while (T--)
  {
    LL f, l;
    cin >> f >> l;
    LL res = noNine(l) - noNine(f) + 1;
    printf("Case #%d: %lld\n", ++o, res);
  }
  return 0;
}